### PR TITLE
Bugfix for bad values in selects

### PR DESCRIPTION
**Pull request by jimmynotjim**
If the select contains an invalid value (an issue we've been seeing often), the scripts will terminate early and log the issue as a tip for developers.

## Additions

- Adds stringValid function and tests
- Adds stringValid checker when sanitizing the list of options

## Testing

- `gulp test:unit:scripts`
- `gulp build` and publish Data & Research's child page Research Reports as well as it's child page Data point: Checking account overdraft then open the page. The error should be logged, but it shouldn't be an error which blocks the rest of the scripts on the page.

## Review

- @anselmbradford 
- @KimberlyMunoz 
- @sebworks 
- @kurtw 

## Notes

- Kurt and I have dealt with bad values more than once, this should help us avoid tearing out our hair if it happens again.

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by sebworks** [comment]
LGTM

File: test/unit_tests/modules/util/strings-spec.js [hunk 1]
```diff
@@ -30,6 +30,30 @@ describe( 'Strings stringEscape()', function() {
   } );
 } );
 
+describe( 'Strings stringValid()', function() {
+  it( 'should return true when testing a valid string', function() {
+    string = 'Test String';
+
+    expect( strings.stringValid( string) ).to.be.true;
```
Comment (KimberlyMunoz) [inline test/unit_tests/modules/util/strings-spec.js]: Trivial spacing after string

**Comment by KimberlyMunoz** [comment]
:+1:

**Comment by jimmynotjim** [comment]
I've added a few new tests to this after @kurtw caught a bug where it wasn't allowing hyphenated words to pass. Could I get a quick additional review if folks have time today? cc @sebworks @KimberlyMunoz

**Comment by KimberlyMunoz** [comment]
:+1:

**Pull request merged**